### PR TITLE
[WIP] Update Sinatra Model Testing Lesson

### PR DESCRIPTION
**Pull request by s-espinosa**
@case-eee This updates the Sinatra model testing lesson to use RSpec and our new Task Manager repo. Let me know what you think!

### CODE REVIEW

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown [hunk 1]
```diff
@@ -69,114 +69,169 @@ run TaskManagerApp
...
-require 'minitest/autorun'
-require 'minitest/pride'
+require 'rack/test'
+require 'rspec'
+require File.expand_path('../../config/environment.rb', __FILE__)
+
+
+module RSpecMixin
+  include Rack::Test::Methods
+  def app() described_class end
```
Comment (case-eee) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: Is this necessary?

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown [hunk 2]
```diff
@@ -69,114 +69,169 @@ run TaskManagerApp
...
-  def test_assigns_attributes_correctly
-    task = Task.new({ "title"       => "a title",
-                      "description" => "a description",
-                      "id"          => 1 })
-    assert_equal "a title", task.title
-    assert_equal "a description", task.description
-    assert_equal 1, task.id
+require_relative '../spec_helper'
+
+RSpec.describe "Test" do
```
Comment (case-eee) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: do we need the `RSpec.describe` or could we just use `describe` here?

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown [hunk 3]
```diff
@@ -69,114 +69,169 @@ run TaskManagerApp
...
+You may have noticed that when we tested the `new` method we were not checking to see if the task we were creating was saved to our database. In order to run that test, we would need to also test our `save` method. Let's add that now.
 
-How will our app know which environment -- test or dev -- we want to use at any moment? By default (like when we start the server with `shotgun`), we will be in development. If we want to run something in the test environment, we need an indicator. We'll use an environment variable: `ENV['RACK_ENV']`. So, in `test/test_helper.rb`:
+In our `/spec/models/task_spec.rb` file, add the following within the RSpec `describe` block:
 
 ```ruby
-ENV['RACK_ENV'] ||= 'test'
+  it 'saves a task to the database' do
```
Comment (case-eee) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: let's put this in a `describe '#save'` block 👍

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown (hunk 2 above)
Comment (s-espinosa) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: @case-eee Based on our discussion, I'm leaving this alone.

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown (hunk 1 above)
Comment (s-espinosa) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: Agree that it's not. Removed now.

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown (hunk 3 above)
Comment (s-espinosa) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: Done! 👍

File: ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown [hunk 4]
```diff
@@ -69,114 +69,162 @@ run TaskManagerApp
...
-Remove any `require` or `require_relative` statements in your controller and/or models except for `require 'yaml/store'` in TaskManager. Our `environment.rb` file now requires all of our controllers, models, and views for us.
+Remove any `require` or `require_relative` statements in your controller and/or models except for `require 'sqlite3'` in Task. Our `environment.rb` file now requires all of our controllers, models, and views for us.
 
 ### Setting up Model Tests
 
-Add `gem 'minitest'` to your Gemfile and then bundle.
+Add the following lines to your `Gemfile`
+
+```ruby
+gem 'rack-test'
```
Comment (case-eee) [inline ruby_02-web_applications_with_ruby/outlines/model_testing_in_sinatra.markdown]: So we probably don't need this as well if we aren't using rack-test anymore 👍 I'm going to go ahead and merge this though and make the update on master!

**Pull request merged**